python/lsst/ap/verify/testPipeline.py

```diff
@@ -0,0 +1,462 @@
+            the detector in question.
+        defects : `lsst.ip.isr.Defects`, optional
+            List of defects.
+        fringes : `lsst.pipe.base.Struct`, optional
+            Struct containing the fringe correction data, with
+            elements:
+            - ``fringes``: fringe calibration frame (`afw.image.Exposure`)
+            - ``seed``: random seed derived from the ccdExposureId for random
+                number generator (`uint32`)
+        opticsTransmission: `lsst.afw.image.TransmissionCurve`, optional
+            A ``TransmissionCurve`` that represents the throughput of the,
+            optics, to be evaluated in focal-plane coordinates.
+        filterTransmission : `lsst.afw.image.TransmissionCurve`
+            A ``TransmissionCurve`` that represents the throughput of the
+            filter itself, to be evaluated in focal-plane coordinates.
+        sensorTransmission : `lsst.afw.image.TransmissionCurve`
+            A ``TransmissionCurve`` that represents the throughput of the
+            sensor itself, to be evaluated in post-assembly trimmed detector
+            coordinates.
+        atmosphereTransmission : `lsst.afw.image.TransmissionCurve`
+            A ``TransmissionCurve`` that represents the throughput of the
+            atmosphere, assumed to be spatially constant.
+        detectorNum : `int`, optional
+            The integer number for the detector to process.
+        isGen3 : bool, optional
+            Flag this call to run() as using the Gen3 butler environment.
+        strayLightData : `object`, optional
+            Opaque object containing calibration information for stray-light
+            correction.  If `None`, no correction will be performed.
+        illumMaskedImage : `lsst.afw.image.MaskedImage`, optional
+            Illumination correction image.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Result struct with components:
+
+            ``exposure``
+                The fully ISR corrected exposure (`afw.image.Exposure`).
+            ``outputExposure``
+                An alias for ``exposure`` (`afw.image.Exposure`).
+            ``ossThumb``
+                Thumbnail image of the exposure after overscan subtraction
+                (`numpy.ndarray`).
+            ``flattenedThumb``
+                Thumbnail image of the exposure after flat-field correction
+                (`numpy.ndarray`).
+        """
+        return Struct(exposure=afwImage.ExposureF(),
+                      outputExposure=afwImage.ExposureF(),
+                      ossThumb=np.empty((1, 1)),
+                      flattenedThumb=np.empty((1, 1)),
+                      )
+
+
+class MockCharacterizeImageTask(PipelineTask):
+    """A do-nothing substitute for CharacterizeImageTask.
+    """
+    ConfigClass = CharacterizeImageConfig
+    _DefaultName = "notCharacterizeImage"
+
+    def __init__(self, butler=None, refObjLoader=None, schema=None, **kwargs):
```

Comment:
I am deliberately imitating [the constructor signature of `CharacterizeImageTask`](https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/characterizeImage.py#L346). I realize that the constructor shouldn't normally be considered part of a class's interface, but in Python it's better safe than sorry -- especially since, as far as I know, Middleware does construct its classes through reflection.

Reply:
Middleware's TaskFactory does not use reflection and it cannot pass parameters that it does not know how to instantiate. TaskFactory only passes three kw params to task constructor: https://github.com/lsst/ctrl_mpexec/blob/main/python/lsst/ctrl/mpexec/taskFactory.py#L99  Extra parameters make no difference as long as they have default values, I think it may make sense to instantiate tasks with non-default params, but this does not happen in middleware.